pkg/controller/complianceremediation/complianceremediation_controller.go

```diff
@@ -300,12 +317,24 @@ func getAppliedMcRemediations(r *ReconcileComplianceRemediation, rem *compv1alph
 
 	appliedRemediations := make([]*mcfgv1.MachineConfig, 0, len(scanSuiteRemediations.Items))
 	for i := range scanSuiteRemediations.Items {
-		if !utils.IsMachineConfig(scanSuiteRemediations.Items[i].Spec.Object) {
+		if !utils.IsMachineConfig(scanSuiteRemediations.Items[i].Spec.Current.Object) {
 			continue
 		}
+
+		var mcObj *unstructured.Unstructured
+
+		// We'll only merge the one that is being reconciled with those that are already
+		// applied. For remediations that are obsolete, we're merge
```

Comment:
Incomplete comment

Reply:
Fixed locally, I'll push the changes all at once to not trigger CI over and over again.